app/src/main/java/de/xikolo/controllers/course/CourseActivity.kt

```diff
@@ -0,0 +1,310 @@
+package de.xikolo.controllers.course
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.design.widget.TabLayout
+import android.support.v4.app.*
+import android.support.v4.view.ViewPager
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewStub
+import android.widget.Button
+import butterknife.BindView
+import com.yatatsu.autobundle.AutoBundleField
+import de.xikolo.R
+import de.xikolo.config.Config
+import de.xikolo.controllers.base.BasePresenterActivity
+import de.xikolo.controllers.dialogs.ProgressDialog
+import de.xikolo.controllers.dialogs.ProgressDialogAutoBundle
+import de.xikolo.controllers.dialogs.UnenrollDialog
+import de.xikolo.controllers.helper.CacheHelper
+import de.xikolo.controllers.helper.CourseArea
+import de.xikolo.controllers.login.LoginActivityAutoBundle
+import de.xikolo.controllers.webview.WebViewFragmentAutoBundle
+import de.xikolo.events.NetworkStateEvent
+import de.xikolo.models.Course
+import de.xikolo.presenters.base.PresenterFactory
+import de.xikolo.presenters.course.CoursePresenter
+import de.xikolo.presenters.course.CoursePresenterFactory
+import de.xikolo.presenters.course.CourseView
+import de.xikolo.utils.ShareUtil
+import de.xikolo.utils.ToastUtil
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+
+class CourseActivity : BasePresenterActivity<CoursePresenter, CourseView>(), CourseView, UnenrollDialog.Listener {
+
+    companion object {
+        val TAG = CourseActivity::class.java.simpleName!!
+    }
+
+    @AutoBundleField(required = true)
+    var courseId: String = "" // workaround, as AutoBundle does not work with Kotlins non-null types
```

Comment:
But you use a non-null type here. `lateinit` should do it, like with butterknife.